evalml/automl/automl_search.py

```diff
@@ -94,6 +94,8 @@ def __init__(self,
                  problem_configuration=None,
                  train_best_pipeline=True,
                  pipeline_parameters=None,
+                 sampler=None,
```

Comment:
Are we ready to make this "public"? If not we should change the name to `_sampler`. If we are ready, perhaps this should be `data_splitting_sampler`?